src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java

```diff
@@ -253,12 +267,12 @@ private void tagSuppressions(Collection<TextBlock> comments) {
     private void tagCommentLine(String text, int line, int column) {
         final Matcher offMatcher = offCommentFormat.matcher(text);
         if (offMatcher.find()) {
-            addTag(offMatcher.group(0), line, column, false);
+            addTag(offMatcher.group(0), line, column, ReportingOptions.NONE);
```

Comment:
this is just another insane code that blow mind on reading.

original true/false was jest a flag in Tag class for ON/OFF type of tags. It is not a reporting option. It is a type of Tag.
There are only two types of Tag - ON and OFF. 
so make TagType {ON, OFF}
so code should be like `offMatcher match results in creation of OFF tag type`:
```
         if (offMatcher.find()) {
            addTag(offMatcher.group(0), line, column, Tag.TagType.OFF); 
```

Tag should be changes to:
```
- private final boolean reportingOn;
+ private final TagType tagType;
```